app/assets/javascripts/components/SlotPicker.es6

```diff
@@ -0,0 +1,119 @@
+(function() {
+  'use strict';
+
+  class SlotPicker extends PWPG.BaseComponent {
+    constructor($component) {
+      super($component);
+
+      this.$months = this.$component.find('.js-slot-picker-month');
+      this.$form = this.$component.parents('.js-day-picker-form');
+      this.selectedClass = 'slot-picker-calendar__action--selected';
+      this.selectedSelector = `.${this.selectedClass}`;
+      this.previousMonthSelector = '.js-slot-picker-prev-month';
+      this.nextMonthSelector = '.js-slot-picker-next-month';
+      this.dayButtonSelector = '.js-day-button';
+      this.timesSelector = '.js-times';
+      this.timesHeaderSelector = '.js-slot-picker-header';
+      this.stepSelector = '#telephone_appointment_step';
+      this.monthSelector = '.js-slot-picker-month';
+      this.errorHeadingSelector = '#error-times-heading';
+      this.slotPickerErrorSelector = '#slot-picker-error';
+      this.slotPickerAjaxLoaderSelector = '#slot-picker-ajax-loader';
+
+      this.init();
+    }
+
+    init() {
+      this.revealMonthButtons();
+      this.selectInitialMonth();
+      this.bindEvents();
+    }
+
+    revealMonthButtons() {
+      $(this.previousMonthSelector).removeClass('hidden');
+      $(this.nextMonthSelector).removeClass('hidden');
+    }
+
+    bindEvents() {
+      $(document).on('click', this.previousMonthSelector, this.handlePrevClick.bind(this));
+      $(document).on('click', this.nextMonthSelector, this.handleNextClick.bind(this));
+      $(document).on('click', this.dayButtonSelector, this.handleDayClick.bind(this));
+    }
+
+    handleDayClick(e) {
+      const $selectedButton = $(e.currentTarget),
+        $currentSelectedButton = this.getSelectedButton(),
+        date = $selectedButton.val();
+
+      e.preventDefault();
+
+      $currentSelectedButton.removeClass(this.selectedClass).attr('disabled', false).attr('aria-pressed', false);
+      $selectedButton.addClass(this.selectedClass).attr('disabled', true).attr('aria-pressed', true);
```

Comment:
I've always used the 2nd actually! - and looking across other JS files in pension guidance this seems to be the pattern used, so have changed it to that.